strawberry/extensions/base_extension.py

```diff
@@ -1,40 +1,33 @@
-from typing import Any, Dict
+from typing import TYPE_CHECKING, Any, Dict
 
 from graphql import GraphQLResolveInfo
 
 from strawberry.types import ExecutionContext
-from strawberry.utils.await_maybe import AwaitableOrValue
+from strawberry.utils.await_maybe import AsyncIteratorOrIterator, AwaitableOrValue
 
 
-class Extension:
-    execution_context: ExecutionContext
-
-    def __init__(self, *, execution_context: ExecutionContext):
-        self.execution_context = execution_context
+# just for getting function nams dynamically and editors auto-complete.
+class _ExtensionHinter:
+    def on_operation(self) -> AsyncIteratorOrIterator[None]:  # pragma: nocover
```

Comment:
This is renamed from `on_request` to `on_operation` since one request can possibly run many graphlq operations and this hook would be called many times.